src/detections/by-days.js

```diff
@@ -0,0 +1,35 @@
+import { data } from 'detections/data.json'
+import { from, reduce } from 'rxjs'
+import { mergeMap } from 'rxjs/operators'
+
+const countDetections = value => {
+  const array = [[], [], [], []]
+
+  value.map(entry => array[Math.floor(new Date(entry.timestamp).getDate() / 8)].push(entry))
```

Comment:
Since it's only this function that changes between `byHours`, `byDays`, etc, let's combine the common logic and only reimplement this function.